### ISSUE TITLE
The file “XY.txt” couldn’t be opened because you don’t have permission to view it.

### ISSUE THREAD

**Issue by leonboe1** [acwright/ImportExport#2]
For me, the app works on the simulator but when I test it on a real device, I will get the error message ```The file “XY.txt” couldn’t be opened because you don’t have permission to view it.``` if I include ```print(error.localizedDescription)``` to the ```catch``` part of the ```.fileImporter```. So importing files doesn't work at all. Exporting files works fine however. (Tested with both iOS 14 and 14.2)

**Comment by acwright** [comment]
Yeah, I have run into this issue as well. I do not believe anything is wrong with the implementation. It may be an Apple sandbox issue?

See #1

**Comment by leonboe1** [comment]
Oh, I didn't notice that the issue had already been posted. However, I found a solution. 

Using CFURLStartAccessingSecurityScopedResource before accessing the file fixes the permission issue. 

But what's weird is that I couldn't find much about this function online. Why are there not more people having the same issue?

Also, it seems like that this function is primarily meant for macOS Development. Maybe it matters for SwiftUI on iOS too because SwiftUI is multi-platform? 

```
if (CFURLStartAccessingSecurityScopedResource(url as CFURL)) {
        
      //Access the url here
        CFURLStopAccessingSecurityScopedResource(url as CFURL)
    }
```

**Comment by acwright** [comment]
Nice! Good find!

I'm more familiar with macOS development than with iOS development and I know that in Xcode for a macOS target this same problem will occur unless you add an entitlement for "User Selected File" which Apple provides a UI for in Xcode. However, there is no similar UI for iOS which is odd. Maybe they haven't got around to that yet? Or maybe there is a way to add that entitlement to iOS?

I have also wondered why others have not had this problem. 

Either way it looks like you have found a workaround. Can you update the code and send a pull request please?

**Comment by acwright** [comment]
Closed by 1641f7c75a4c367623591e9ede8236bf94cf4ecd
